scripts/datasets/coco_tracking.py

```diff
@@ -0,0 +1,242 @@
+    parser.add_argument('--num-threads', type=int, default=12,
+                        help='threads number')
+    args = parser.parse_args()
+    args.download_dir = expanduser(args.download_dir)
+    return args
+
+def download_coco(args, overwrite=False):
+    """download COCO dataset and Unzip to download_dir"""
+    _DOWNLOAD_URLS = [
+        ('http://images.cocodataset.org/zips/train2017.zip',
+         '10ad623668ab00c62c096f0ed636d6aff41faca5'),
+        ('http://images.cocodataset.org/annotations/annotations_trainval2017.zip',
+         '8551ee4bb5860311e79dace7e79cb91e432e78b3'),
+        ('http://images.cocodataset.org/zips/val2017.zip',
+         '4950dc9d00dbe1c933ee0170f5797584351d2a41'),
+    ]
+    if not isdir(args.download_dir):
+        makedirs(args.download_dir)
+    for url, checksum in _DOWNLOAD_URLS:
+        filename = download(url, path=args.download_dir, overwrite=overwrite, sha1_hash=checksum)
+        with zipfile.ZipFile(filename) as zf:
+            zf.extractall(path=args.download_dir)
+
+def printProgress(iteration, total, prefix='', suffix='', decimals=1, barLength=100):
+    """
+    Call in a loop to create terminal progress bar
+
+    Parameters
+    ----------
+        iteration: int : current iteration.
+        total: int, total iterations.
+        prefix: str, prefix string.
+        suffix: str, suffix string.
+        decimals: int, positive number of decimals in percent complete.
+        barLength: int, character length of bar.
+    """
+    formatStr = "{0:." + str(decimals) + "f}"
+    percents = formatStr.format(100 * (iteration / float(total)))
+    filledLength = int(round(barLength * iteration / float(total)))
+    bar = '' * filledLength + '-' * (barLength - filledLength)
+    sys.stdout.write('\r%s |%s| %s%s %s' % (prefix, bar, percents, '%', suffix)),
+    if iteration == total:
+        sys.stdout.write('\x1b[2K\r')
+    sys.stdout.flush()
+
+
+def crop_hwc(image, bbox, out_sz, padding=(0, 0, 0)):
+    """
+    crop image
+
+    Parameters
+    ----------
+    image: np.array, image
+    bbox: np or list, bbox coordinate [xmin,ymin,xmax,ymax]
+    out_sz: int , crop image size
+
+    Return:
+        crop result
+    """
+    a = (out_sz-1) / (bbox[2]-bbox[0])
+    b = (out_sz-1) / (bbox[3]-bbox[1])
+    c = -a * bbox[0]
+    d = -b * bbox[1]
+    mapping = np.array([[a, 0, c],
+                        [0, b, d]]).astype(np.float)
+    crop = cv2.warpAffine(image, mapping, (out_sz, out_sz), borderMode=cv2.BORDER_CONSTANT, borderValue=padding)
+    return crop
+
+
+def pos_s_2_bbox(pos, s):
+    """
+    from center_x,center_y,s to get bbox
+
+    Parameters
+    ----------
+    pos , x, bbox
+    s , int, bbox size
+
+    Return:
+        [x_min,y_min,x_max,y_max]
+    """
+    return [pos[0]-s/2, pos[1]-s/2, pos[0]+s/2, pos[1]+s/2]
+
+
+def crop_like_SiamFC(image, bbox, context_amount=0.5, exemplar_size=127, instance_size=255, padding=(0, 0, 0)):
```

Comment:
is the cropping function reusable? If so you refactor the duplicate function into gluoncv.utils.data?

Reply:
Yes, many functions are reusable. Yukun mentioned this to me yesterday, but we haven't decided where is a good location.

It seems there is no `data.py` file or `data` class under `gluoncv/utils`. Should we have a new script named `data.py` or `tracking.py` under `gluoncv/utils`?

Reply:
gluoncv/utils/data/tracking.py

Reply:
Got it.